src/navigator/header/CreateCommunity.tsx

```diff
@@ -2,17 +2,24 @@ import { StackNavigationProp } from '@react-navigation/stack';
 import i18n from 'assets/i18n';
 import { Screens } from 'helpers/constants';
 import { CommunityAttributes } from 'helpers/types/models';
+import { IRootState } from 'helpers/types/state';
 import React from 'react';
 import { View } from 'react-native';
 import { Button } from 'react-native-paper';
+import { useSelector } from 'react-redux';
 import { ipctColors } from 'styles/index';
 
 function CreateCommunity(props: {
     navigation: StackNavigationProp<any, any>;
     userCommunity?: CommunityAttributes;
 }) {
+    const walletAddress = useSelector(
+        (state: IRootState) => state.user.wallet.address
+    );
+    // TODO: create should always be visible, but allowed only after login
     return (
-        !props.userCommunity && (
+        !props.userCommunity &&
+        walletAddress.length > 0 && (
```

Comment:
Is the double-check needed? Isn't enough to check for a userCommunity? Only logged users will have one.

Reply:
Create should only be visible when the user has no community. Yes, non-logged-in users, do not have a community, so they should be able to see the button. But then, entering the "create community" view, the wallet address is needed to create the community. If they are not logged in, they will not be able to create a community, so we should show them a message saying that they need to login. Likely, you've removed that by mistake. Since that was a bit more work, I've just added this condition to close the version. I've also added a task to fix that later https://impactmarket.atlassian.net/browse/IPCT1-244